### ISSUE TITLE
source-mongodb-v2 failing to make incremental read with "resume token was not found"

### ISSUE THREAD

**Issue by rodireich** [airbytehq/airbyte#33790]
### Connector Name

source-mongodb-v2

### Connector Version

1.2.1

### What step the error happened?

During the sync

### Relevant information

OSS customer running on their own eke cluster reached out.
Using mongodb-v2 source. A single table out of many failing on incremental sync.
Initial sync is reading correctly all 7m documents / 13GB. But then incremental attempt is failing with "resume token was not found".
There is nothing that appears to be special about this table 

Using mongodb on Atlas. The customer confirmed this resume token is found at the time we're failing.

[[thread]](https://airbytehq-team.slack.com/archives/C043JHEEYKG/p1703625420052309)

### Relevant log output

```shell
2023-12-26 21:10:41 source > com.mongodb.MongoCommandException: Command failed with error 280 (ChangeStreamFatalError): 'PlanExecutor error during aggregation :: caused by :: cannot resume stream; the resume token was not found. {_data: "82658B3ED8000000012B022C0100296E5A10045E66C3ACFD4C48E08BD30480813E5BA9463C5F6964003C64393434613133393736363061653362653932646235386639653061626566336461353466343733626162343431633635323735666433333962653231353438000004"}' on server finaloop-prod-shard-00-01.xnbzo.mongodb.net:27017. The full response is {"errorLabels": ["NonResumableChangeStreamError"], "ok": 0.0, "errmsg": "PlanExecutor error during aggregation :: caused by :: cannot resume stream; the resume token was not found. {_data: \"82658B3ED8000000012B022C0100296E5A10045E66C3ACFD4C48E08BD30480813E5BA9463C5F6964003C64393434613133393736363061653362653932646235386639653061626566336461353466343733626162343431633635323735666433333962653231353438000004\"}", "code": 280, "codeName": "ChangeStreamFatalError", "$clusterTime": {"clusterTime": {"$timestamp": {"t": 1703625041, "i": 31}}, "signature": {"hash": {"$binary": {"base64": "nSPEp5HQR8fsetHepz2FCenGf+E=", "subType": "00"}}, "keyId": 7270901963045208066}}, "operationTime": {"$timestamp": {"t": 1703625041, "i": 12}}}
2023-12-26 21:10:41 source > 	at com.mongodb.internal.connection.ProtocolHelper.getCommandFailureException(ProtocolHelper.java:205) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.connection.InternalStreamConnection.receiveCommandMessageResponse(InternalStreamConnection.java:448) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.connection.InternalStreamConnection.sendAndReceive(InternalStreamConnection.java:370) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.connection.UsageTrackingInternalConnection.sendAndReceive(UsageTrackingInternalConnection.java:114) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.connection.DefaultConnectionPool$PooledConnection.sendAndReceive(DefaultConnectionPool.java:719) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.connection.CommandProtocolImpl.execute(CommandProtocolImpl.java:76) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.connection.DefaultServer$DefaultServerProtocolExecutor.execute(DefaultServer.java:203) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.connection.DefaultServerConnection.executeProtocol(DefaultServerConnection.java:115) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.connection.DefaultServerConnection.command(DefaultServerConnection.java:83) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.connection.DefaultServerConnection.command(DefaultServerConnection.java:74) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.connection.DefaultServer$OperationCountTrackingConnection.command(DefaultServer.java:287) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.operation.CommandOperationHelper.createReadCommandAndExecute(CommandOperationHelper.java:245) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.operation.CommandOperationHelper.lambda$executeRetryableRead$6(CommandOperationHelper.java:227) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.operation.OperationHelper.lambda$withSourceAndConnection$0(OperationHelper.java:345) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.operation.OperationHelper.withSuppliedResource(OperationHelper.java:370) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.operation.OperationHelper.lambda$withSourceAndConnection$1(OperationHelper.java:344) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.operation.OperationHelper.withSuppliedResource(OperationHelper.java:370) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.operation.OperationHelper.withSourceAndConnection(OperationHelper.java:343) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.operation.CommandOperationHelper.lambda$executeRetryableRead$7(CommandOperationHelper.java:225) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.operation.CommandOperationHelper.lambda$decorateReadWithRetries$3(CommandOperationHelper.java:192) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.async.function.RetryingSyncSupplier.get(RetryingSyncSupplier.java:67) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.operation.CommandOperationHelper.executeRetryableRead(CommandOperationHelper.java:230) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.operation.CommandOperationHelper.executeRetryableRead(CommandOperationHelper.java:212) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.operation.AggregateOperationImpl.execute(AggregateOperationImpl.java:191) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.operation.ChangeStreamOperation.lambda$execute$0(ChangeStreamOperation.java:187) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.operation.OperationHelper.withReadConnectionSource(OperationHelper.java:308) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.operation.ChangeStreamOperation.execute(ChangeStreamOperation.java:185) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.internal.operation.ChangeStreamOperation.execute(ChangeStreamOperation.java:55) ~[mongodb-driver-core-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.client.internal.MongoClientDelegate$DelegateOperationExecutor.execute(MongoClientDelegate.java:153) ~[mongodb-driver-sync-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.client.internal.ChangeStreamIterableImpl.execute(ChangeStreamIterableImpl.java:212) ~[mongodb-driver-sync-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at com.mongodb.client.internal.ChangeStreamIterableImpl.cursor(ChangeStreamIterableImpl.java:187) ~[mongodb-driver-sync-4.10.2.jar:?]
2023-12-26 21:10:41 source > 	at io.debezium.connector.mongodb.MongoDbStreamingChangeEventSource.readChangeStream(MongoDbStreamingChangeEventSource.java:168) ~[debezium-connector-mongodb-2.4.0.Final.jar:2.4.0.Final]
2023-12-26 21:10:41 source > 	at io.debezium.connector.mongodb.MongoDbStreamingChangeEventSource.lambda$streamChangesForReplicaSet$0(MongoDbStreamingChangeEventSource.java:99) ~[debezium-connector-mongodb-2.4.0.Final.jar:2.4.0.Final]
2023-12-26 21:10:41 source > 	at io.debezium.connector.mongodb.connection.MongoDbConnection.lambda$execute$1(MongoDbConnection.java:94) ~[debezium-connector-mongodb-2.4.0.Final.jar:2.4.0.Final]
2023-12-26 21:10:41 source > 	at io.debezium.connector.mongodb.connection.MongoDbConnection.execute(MongoDbConnection.java:110) ~[debezium-connector-mongodb-2.4.0.Final.jar:2.4.0.Final]
2023-12-26 21:10:41 source > 	at io.debezium.connector.mongodb.connection.MongoDbConnection.execute(MongoDbConnection.java:93) ~[debezium-connector-mongodb-2.4.0.Final.jar:2.4.0.Final]
2023-12-26 21:10:41 source > 	at io.debezium.connector.mongodb.MongoDbStreamingChangeEventSource.streamChangesForReplicaSet(MongoDbStreamingChangeEventSource.java:98) ~[debezium-connector-mongodb-2.4.0.Final.jar:2.4.0.Final]
2023-12-26 21:10:41 source > 	at io.debezium.connector.mongodb.MongoDbStreamingChangeEventSource.execute(MongoDbStreamingChangeEventSource.java:82) ~[debezium-connector-mongodb-2.4.0.Final.jar:2.4.0.Final]
2023-12-26 21:10:41 source > 	at io.debezium.connector.mongodb.MongoDbStreamingChangeEventSource.execute(MongoDbStreamingChangeEventSource.java:41) ~[debezium-connector-mongodb-2.4.0.Final.jar:2.4.0.Final]
2023-12-26 21:10:41 source > 	at io.debezium.pipeline.ChangeEventSourceCoordinator.streamEvents(ChangeEventSourceCoordinator.java:272) ~[debezium-core-2.4.0.Final.jar:2.4.0.Final]
2023-12-26 21:10:41 source > 	at io.debezium.pipeline.ChangeEventSourceCoordinator.executeChangeEventSources(ChangeEventSourceCoordinator.java:197) ~[debezium-core-2.4.0.Final.jar:2.4.0.Final]
2023-12-26 21:10:41 source > 	at io.debezium.pipeline.ChangeEventSourceCoordinator.lambda$start$0(ChangeEventSourceCoordinator.java:137) ~[debezium-core-2.4.0.Final.jar:2.4.0.Final]
2023-12-26 21:10:41 source > 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) [?:?]
2023-12-26 21:10:41 source > 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
2023-12-26 21:10:41 source > 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
2023-12-26 21:10:41 source > 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
2023-12-26 21:10:41 source > 	at java.base/java.lang.Thread.run(Thread.java:833) [?:?]
2023-12-26 21:10:41 source > 2023-12-26 21:10:41 ERROR i.d.c.m.MongoDbStreamingChangeEventSource(streamChangesForReplicaSet):103 - Streaming for replica set atlas-elry3t-shard-0 failed
2023-12-26 21:10:41 source > io.debezium.DebeziumException: Error while attempting to read from change stream on 'atlas-elry3t-shard-0'
2023-12-26 21:10:41 source > 	at io.debezium.connector.mongodb.MongoDbChangeEventSourceFactory.lambda$connectionErrorHandler$1(MongoDbChangeEventSourceFactory.java:126) ~[debezium-connector-mongodb-2.4.0.Final.jar:2.4.0.Final]
2023-12-26 21:10:41 source > 	at io.debezium.connector.mongodb.connection.MongoDbConnection.execute(MongoDbConnection.java:116) ~[debezium-connector-mongodb-2.4.0.Final.jar:2.4.0.Final]
2023-12-26 21:10:41 source > 	at io.debezium.connector.mongodb.connection.MongoDbConnection.execute(MongoDbConnection.java:93) ~[debezium-connector-mongodb-2.4.0.Final.jar:2.4.0.Final]
2023-12-26 21:10:41 source > 	at io.debezium.connector.mongodb.MongoDbStreamingChangeEventSource.streamChangesForReplicaSet(MongoDbStreamingChangeEventSource.java:98) ~[debezium-connector-mongodb-2.4.0.Final.jar:2.4.0.Final]
2023-12-26 21:10:41 source > 	at io.debezium.connector.mongodb.MongoDbStreamingChangeEventSource.execute(MongoDbStreamingChangeEventSource.java:82) ~[debezium-connector-mongodb-2.4.0.Final.jar:2.4.0.Final]
2023-12-26 21:10:41 source > 	at io.debezium.connector.mongodb.MongoDbStreamingChangeEventSource.execute(MongoDbStreamingChangeEventSource.java:41) ~[debezium-connector-mongodb-2.4.0.Final.jar:2.4.0.Final]
2023-12-26 21:10:41 source > 	at io.debezium.pipeline.ChangeEventSourceCoordinator.streamEvents(ChangeEventSourceCoordinator.java:272) ~[debezium-core-2.4.0.Final.jar:2.4.0.Final]
2023-12-26 21:10:41 source > 	at io.debezium.pipeline.ChangeEventSourceCoordinator.executeChangeEventSources(ChangeEventSourceCoordinator.java:197) ~[debezium-core-2.4.0.Final.jar:2.4.0.Final]
2023-12-26 21:10:41 source > 	at io.debezium.pipeline.ChangeEventSourceCoordinator.lambda$start$0(ChangeEventSourceCoordinator.java:137) ~[debezium-core-2.4.0.Final.jar:2.4.0.Final]
2023-12-26 21:10:41 source > 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) [?:?]
2023-12-26 21:10:41 source > 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
2023-12-26 21:10:41 source > 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
2023-12-26 21:10:41 source > 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
2023-12-26 21:10:41 source > 	at java.base/java.lang.Thread.run(Thread.java:833) [?:?]
```

### Contribute

- [ ] Yes, I want to contribute

**Comment by rodireich** [comment]
Related though may not be the same issue: https://github.com/airbytehq/oncall/issues/3675

**Comment by klsoper** [comment]
adding affected user:


| Environment | Affected Workspaces/Customer | Connection Link | Case/Chat Link | Sentiment | User Data Access (no longer manditory, [see guidelines](https://www.notion.so/Accessing-Customer-Data-ceaf710c5ebe451b92d69ddbce1c3bfc_)) | Reporter |
| ---- | ---- | ------------ | --------- | ---------- | ---------- | ---------- |
| cloud | joao.machado@euentrego.com | [mongo <> clickhouse ](https://cloud.airbyte.com/workspaces/47d73fbc-f898-4a64-9d0e-539b3d5fd3d6/connections/0f197f29-1dd7-431d-afa3-5af3cf8e0161/job-history) | [ticket](https://airbyte1416.zendesk.com/agent/tickets/3331) |  :ok:  | Y/N | @klsoper  |

**Comment by klsoper** [comment]
@rodireich is there an eta on a fix here?

**Comment by klsoper** [comment]
Not sure how to interpret what they replied: The entire table is some 10MB or 30K records.
How much accesss does it have?

asking about access now

**Comment by rodireich** [comment]
Thanks. That would be super helpful

**Comment by klsoper** [comment]
@rodireich reply from user 

Here are the collection stats.

STORAGE SIZE: 1.92MB
LOGICAL DATA SIZE: 6.04MB
TOTAL DOCUMENTS: 30839
INDEXES TOTAL SIZE: 1.28MB


I tried with bigger collections but now I'm sugin this small one. The error is the same.

**Comment by AronsonDan** [comment]
Hey @rodireich ,

I was trying to take a different approach.

Took 13 different collections and inserted them into on connection.
attaching the logs:

**Comment by octavia-squidington-iii** [comment]
Zendesk ticket #[3331](https://airbyte1416.zendesk.com/agent/tickets/3331) has been linked to this issue.

**Comment by erica-airbyte** [comment]
@rodireich any update here?

**Comment by klsoper** [comment]
@rodireich can I get an update on this please?

**Comment by rodireich** [comment]
@klsoper This should be resolved now with 1.2.6

**Comment by AronsonDan** [comment]
@rodireich 🍾
